app/controllers/api/v1/logs_controller.rb

```diff
@@ -12,4 +12,21 @@ def cal_heatmap
     logs = user.logs.where(created_date: params[:start]..params[:stop]).map { |c| c.created_date.to_time.to_i }
     render json: logs.each_with_object(Hash.new(0)) { |tm, h| h[tm] += 1 }
   end
+
+  def graph
```

Comment:
__[rails_best_practices]__
* move model logic into model (user use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)
* move model logic into model ( use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)